pki/pki.go

```diff
@@ -29,287 +28,117 @@ type CertificatePKI struct {
-	}
+	certs[KubeAPICertName] = ToCertObject(KubeAPICertName, "", "", kubeAPICrt, kubeAPIKey)
 
 	// generate Kube controller-manager certificate and key
 	log.Infof(ctx, "[certificates] Generating Kube Controller certificates")
-	kubeControllerCrt, kubeControllerKey, err := generateClientCertAndKey(caCrt, caKey, KubeControllerCommonName, []string{})
+	kubeControllerCrt, kubeControllerKey, err := GenerateSignedCertAndKey(caCrt, caKey, nil, nil, getDefaultCN(KubeControllerCertName), nil, false)
 	if err != nil {
 		return nil, err
 	}
-	logrus.Debugf("[certificates] Kube Controller Certificate: %s", string(cert.EncodeCertPEM(kubeControllerCrt)))
-	certs[KubeControllerName] = CertificatePKI{
-		Certificate:   kubeControllerCrt,
-		Key:           kubeControllerKey,
-		Config:        getKubeConfigX509("https://127.0.0.1:6443", KubeControllerName, CACertPath, KubeControllerCertPath, KubeControllerKeyPath),
-		Name:          KubeControllerName,
-		CommonName:    KubeControllerCommonName,
-		EnvName:       KubeControllerCertENVName,
-		KeyEnvName:    KubeControllerKeyENVName,
-		Path:          KubeControllerCertPath,
-		KeyPath:       KubeControllerKeyPath,
-		ConfigEnvName: KubeControllerConfigENVName,
-		ConfigPath:    KubeControllerConfigPath,
-	}
+	certs[KubeControllerCertName] = ToCertObject(KubeControllerCertName, "", "", kubeControllerCrt, kubeControllerKey)
 
 	// generate Kube scheduler certificate and key
 	log.Infof(ctx, "[certificates] Generating Kube Scheduler certificates")
-	kubeSchedulerCrt, kubeSchedulerKey, err := generateClientCertAndKey(caCrt, caKey, KubeSchedulerCommonName, []string{})
+	kubeSchedulerCrt, kubeSchedulerKey, err := GenerateSignedCertAndKey(caCrt, caKey, nil, nil, getDefaultCN(KubeSchedulerCertName), nil, false)
 	if err != nil {
 		return nil, err
 	}
-	logrus.Debugf("[certificates] Kube Scheduler Certificate: %s", string(cert.EncodeCertPEM(kubeSchedulerCrt)))
-	certs[KubeSchedulerName] = CertificatePKI{
-		Certificate:   kubeSchedulerCrt,
-		Key:           kubeSchedulerKey,
-		Config:        getKubeConfigX509("https://127.0.0.1:6443", KubeSchedulerName, CACertPath, KubeSchedulerCertPath, KubeSchedulerKeyPath),
-		Name:          KubeSchedulerName,
-		CommonName:    KubeSchedulerCommonName,
-		EnvName:       KubeSchedulerCertENVName,
-		KeyEnvName:    KubeSchedulerKeyENVName,
-		Path:          KubeSchedulerCertPath,
-		KeyPath:       KubeSchedulerKeyPath,
-		ConfigEnvName: KubeSchedulerConfigENVName,
-		ConfigPath:    KubeSchedulerConfigPath,
-	}
+	certs[KubeSchedulerCertName] = ToCertObject(KubeSchedulerCertName, "", "", kubeSchedulerCrt, kubeSchedulerKey)
 
 	// generate Kube Proxy certificate and key
 	log.Infof(ctx, "[certificates] Generating Kube Proxy certificates")
-	kubeProxyCrt, kubeProxyKey, err := generateClientCertAndKey(caCrt, caKey, KubeProxyCommonName, []string{})
+	kubeProxyCrt, kubeProxyKey, err := GenerateSignedCertAndKey(caCrt, caKey, nil, nil, getDefaultCN(KubeProxyCertName), nil, false)
```

Comment:
Nit: when generate method signature, try to put  params that are most likely will not be nil, first.